src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -171,7 +171,32 @@ public LanguageVersion LanguageVersion
 
         public override INamedTypeSymbol CreateErrorTypeSymbol(INamespaceOrTypeSymbol container, string name, int arity)
         {
-            return new ExtendedErrorTypeSymbol((NamespaceOrTypeSymbol)container, name, arity, null);
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (arity < 0)
+            {
+                throw new ArgumentException(nameof(arity));
+            }
+
+            return new ExtendedErrorTypeSymbol(container.EnsureCSharpSymbolOrNull<INamespaceOrTypeSymbol, NamespaceOrTypeSymbol>(nameof(container)), name, arity, null);
```

Comment:
named parameter for 'null'.

Nit: Wrap long lines to prevent having to scroll horizontally in github.